tests/unit/protocol-details.js

```diff
@@ -0,0 +1,252 @@
+/* global describe, beforeEach, afterEach, it, expect, spyOn */
+
+describe('protocol-details.js', function() {
+
+    'use strict';
+
+    beforeEach(function () {
+        var details = '<div id="test-wrapper">' +
+                          '<section id="test1-details">' +
+                              '<h3 id="test1-heading-1">Heading 1</h3>' +
+                              '<p>Paragraph 1</p>' +
+                              '<h3 id="test1-heading-2">Heading 2</h3>' +
+                              '<p>Paragraph 2</p>' +
+                              '<h4 id="test1-sub-heading-1">Sub Heading 1</h4>' +
+                              '<p>Paragraph 3</p>' +
+                           '</section>' +
+                           '<section id="test2-details">' +
+                              '<h3 id="test2-heading-1">Heading 1</h3>' +
+                              '<p>Paragraph 1</p>' +
+                              '<h3 id="test2-heading-2">Heading 2</h3>' +
+                              '<p>Paragraph 2</p>' +
+                              '<h4 id="test2-sub-heading-1">Sub Heading 1</h4>' +
+                              '<p>Paragraph 3</p>' +
+                           '</section>'+
+                      '</div>';
+
+        document.documentElement.insertAdjacentHTML('beforeend', details);
+    });
+
+    afterEach(function(){
+        var node = document.querySelector('#test-wrapper');
+        if (node.parentNode) {
+            node.parentNode.removeChild(node);
+        }
+    });
+
+    describe('init', function () {
+
+        it('has Utils', function() {
+            expect(Mzp.Utils.nextUntil).toBeTruthy();
+        });
+
+        it('does not initalize if the browser does not support it', function() {
+            Mzp.Details.isSupported = false;
```

Comment:
The function is in a closure so it only run the first time the file loads. It doesn't run with each test so threre's nothing to spy on.

Reply:
Ok, in that case we should at least make sure that the value you're overriding gets reset back to the browser's initial value after each test, so we don't need to rely on things like the order of the tests in order to get everything to pass. Alternatively, you could just make this a simple function and not a closure. I don't think there would be much of a perf hit here since it's only being called once on `init()`.

Reply:
OMG THANK YOU. I spent 5 hours on this on Friday.

Reply:
Okay, I took this out of the closure and update the tests to spy on it.